nemo/collections/nlp/models/intent_slot_classification/intent_slot_classification_model.py

```diff
@@ -53,70 +53,124 @@ def output_types(self) -> Optional[Dict[str, NeuralType]]:
     def __init__(self, cfg: DictConfig, trainer: Trainer = None):
         """ Initializes BERT Joint Intent and Slot model.
         """
-
-        self.data_dir = cfg.data_dir
         self.max_seq_length = cfg.language_model.max_seq_length
 
-        self.data_desc = IntentSlotDataDesc(
-            data_dir=cfg.data_dir, modes=[cfg.train_ds.prefix, cfg.validation_ds.prefix]
-        )
+        # Check the presence of data_dir.
+        if not cfg.data_dir:
+            # Disable setup methods.
+            IntentSlotClassificationModel._set_model_restore_state(is_being_restored=True)
+        else:
+            # Update configuration with new data.
+            self._copy_data_desc_to_cfg(cfg, cfg.data_dir, cfg.train_ds, cfg.validation_ds)
+            # Conditional initialization of tokenizer.
+            self.data_dir = cfg.data_dir
+            self._setup_tokenizer(cfg.tokenizer)
 
-        self._setup_tokenizer(cfg.tokenizer)
         # init superclass
         super().__init__(cfg=cfg, trainer=trainer)
 
+        # Check the presence of data_dir.
+        if cfg.data_dir:
+            # Conditional initialization of modules.
+            self._init_modules_when_tokenizer_set()
+
+        # Enable setup methods.
+        IntentSlotClassificationModel._set_model_restore_state(is_being_restored=False)
```

Comment:
Why do we need it here?

Reply:
to disable calling setup when `cfg.data_dir` is not set